src/javascript/app/Stores/Modules/Trading/trade-store.js

```diff
@@ -555,22 +555,30 @@ export default class TradeStore extends BaseStore {
 
     @action.bound
     async onMount() {
+        this.onLoadingMount();
         await this.prepareTradeStore();
         this.debouncedProposal();
         runInAction(() => {
             this.is_trade_component_mounted = true;
-            this.onLoadingMount();
         });
         this.updateQueryString();
         this.onSwitchAccount(this.accountSwitcherListener);
     }
 
     @action.bound
     onLoadingMount() {
-        // TODO: find better way to remove initial loader
-        setTimeout(() => {
-            this.root_store.ui.setAppLoading(false);
-        }, 2200);
+        let time_counter = 0;
+        const loading_interval = setInterval(() => {
+            if (this.smart_chart.is_chart_ready && this.is_trade_component_mounted) {
+                this.root_store.ui.setAppLoading(false);
+                clearInterval(loading_interval);
+            }
+            if (time_counter === 20) { // 20 iterations = 400 * 20 = 8 seconds
```

Comment:
we can use a setTimeout wrapped around the setInterval instead

Reply:
when using timeout wrapped in setinterval, even after setinterval done and cleartimeout the timeout will always run, therefore, it needs to be seperated